src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -277,7 +277,13 @@ private IEnumerable<DiagnosticAnalyzer> GetAnalyzersForTypeNames(Assembly analyz
                 Type type;
                 try
                 {
-                    type = Type.GetType(typeName + ", " + analyzerAssembly.FullName, throwOnError: true);
+                    // TODO: Once we move to CoreCLR we should just call GetType(typeName, throwOnError: true, ignoreCase: false) directly.
+                    // For now we fall back to reflection shim in order to report good error message (type load exception).
+                    const bool throwOnError = true;
+                    const bool ignoreCase = false;
+
+                    type = analyzerAssembly.GetType(typeName) ?? 
+                           PortableShim.Assembly.GetType_string_bool_bool(analyzerAssembly, typeName, throwOnError, ignoreCase);
```

Comment:
@tmat i think the reflection cost would be low.  We pay the one time cost to get the method but after that the `CreateDelegate` call makes it essentially 0 cost.